os/net/netmgr/netmgr_ioctl_lwip.c

```diff
@@ -59,6 +59,31 @@ static struct netif *_netdev_dhcp_dev(FAR const char *intf)
 }
 #endif
 
+static int _netdev_set_dnsserver(struct sockaddr *addr, int index)
+{
+	ip_addr_t dns_addr;
+	struct sockaddr_in *iaddr = (struct sockaddr_in *)addr;
+	ip_addr_set_ip4_u32(&dns_addr, iaddr->sin_addr.s_addr);
+	if (index > 0) {
+		if (index >= CONFIG_NET_DNS_MAX_SERVERS) {
+			return -2;
+		} else {
+			dns_setserver(index, &dns_addr);
+			return 0;
+		}
+	}
+	/* Set DNS server to available slot.*/
+	for (int i = 0; i < CONFIG_NET_DNS_MAX_SERVERS; i++) {
```

Comment:
Is it possible to please declare i at the start of the function separately?

Reply:
it's possible but I don't think it's necessary.